dumpling/export/status.go

```diff
@@ -29,25 +28,41 @@ func (d *Dumper) runLogProgress(tctx *tcontext.Context) {
 		case <-logProgressTicker.C:
 			nanoseconds := float64(time.Since(lastCheckpoint).Nanoseconds())
 
-			completedTables := ReadCounter(finishedTablesCounter, conf.Labels)
-			finishedBytes := ReadGauge(finishedSizeGauge, conf.Labels)
-			finishedRows := ReadGauge(finishedRowsGauge, conf.Labels)
-			estimateTotalRows := ReadCounter(estimateTotalRowsCounter, conf.Labels)
-
 			tctx.L().Info("progress",
-				zap.String("tables", fmt.Sprintf("%.0f/%.0f (%.1f%%)", completedTables, totalTables, completedTables/totalTables*100)),
-				zap.String("finished rows", fmt.Sprintf("%.0f", finishedRows)),
-				zap.String("estimate total rows", fmt.Sprintf("%.0f", estimateTotalRows)),
-				zap.String("finished size", units.HumanSize(finishedBytes)),
-				zap.Float64("average speed(MiB/s)", (finishedBytes-lastBytes)/(1048576e-9*nanoseconds)),
+				zap.String("tables", fmt.Sprintf("%.0f/%.0f (%.1f%%)", midd.CompletedTables, midd.TotalTables, midd.CompletedTables/midd.TotalTables*100)),
+				zap.String("finished rows", fmt.Sprintf("%.0f", midd.FinishedRows)),
+				zap.String("estimate total rows", fmt.Sprintf("%.0f", midd.EstimateTotalRows)),
+				zap.String("finished size", units.HumanSize(midd.FinishedBytes)),
+				zap.Float64("average speed(MiB/s)", (midd.FinishedBytes-lastBytes)/(1048576e-9*nanoseconds)),
 			)
 
 			lastCheckpoint = time.Now()
-			lastBytes = finishedBytes
+			lastBytes = midd.FinishedBytes
 		}
 	}
 }
 
+type Midparams struct {
+	TotalTables       float64
+	CompletedTables   float64
+	FinishedBytes     float64
+	FinishedRows      float64
+	EstimateTotalRows float64
+}
+
+func (d *Dumper) GetParameters() (midparams *Midparams) {
+	conf := d.conf
+	mid := &Midparams{}
+	if mid.TotalTables == 0 {
+		mid.TotalTables = float64(calculateTableCount(conf.Tables))
+	}
```

Comment:
emmm. This doesn't make any change, it will still be computed every time. Maybe you can store a integer in `Dumper` and set it only once(maybe after `conf.Tables` is generated)